### ISSUE TITLE
Header - Front end

### ISSUE THREAD

**Issue by iamskip** [GemeenteNijmegen/component-library#194]
**Doel**
Het doel van dit issue is om het front-end Header-component te ontwikkelen op basis van het ontwerp. Dit component zorgt voor de presentatie van de primaire navigatie en merkidentiteit op een visueel aantrekkelijke en toegankelijke manier.

**Beschrijving**
Het Header-component zal worden ontwikkeld om de primaire navigatie, logo, secundaire elementen zoals een zoekfunctie en taalkeuze te bevatten. Het component moet responsive zijn en aansluiten bij de bestaande visuele en technische richtlijnen.

**Acceptatiecriteria (Acc):**
	•	Het Header-component moet de functionaliteit bieden zoals beschreven in het ontwerp.
	•	Het Header-component moet verschillende variaties ondersteunen, waaronder:
	•	Standaard Header met logo en primaire navigatie
	•	Header met optionele componenten zoals zoekfunctie of taalkeuze
	•	Het component moet visueel consistent zijn met andere componenten in het systeem.
	•	Het Header-component moet toegankelijk zijn, met voldoende kleurcontrast en ondersteuning voor toetsenbordnavigatie en schermlezers.
	•	Het component moet responsive zijn en goed werken op verschillende schermgroottes.

**Definitie van ‘Done’ (DoD):**
	•	Het Header-component is volledig ontwikkeld volgens de specificaties in het ontwerp.
	•	Het component is getest en werkt correct op verschillende schermformaten en browsers.
	•	Eventuele feedback van het team is verwerkt.
	•	De ontwikkelde code is gedocumenteerd en toegevoegd aan Storybook.
	•	Het component voldoet aan alle acceptatiecriteria, inclusief toegankelijkheid.

**Comment by Michel-Verhoeven** [comment]
[De header inclusief mega menu en toolbar button op mobile view staat op staging](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-header--default) en kan getest worden

**Comment by Rozerinay** [comment]
Ziet er al goed uit! Ik heb nog een paar opmerkingen:

- De Header schakelt al bij een schermbreedte van 1100px naar een andere weergave. Hierdoor komen de menu-items te dicht op elkaar te staan, en worden sommige woorden naar een volgende regel geduwd (zie screenshots). In het [Schetsboek](https://www.figma.com/design/OnbEtMvfeOF74vXn9ayZnA/Schetsboek---Gemeente-Nijmegen?node-id=403-3208&t=F8tILOmEbCaDFzqr-11) heb ik aangegeven dat deze overgang beter vanaf het breakpoint van 1280px kan plaatsvinden.
- Als het niet wenselijk is om deze overgang pas bij 1280px te laten starten, kunnen we eventueel een alternatieve schets aanleveren voor een tussenoplossing, bijvoorbeeld met een compactere weergave van ‘Mijn Nijmegen’.

**Comment by Rozerinay** [comment]
@Michel-Verhoeven ik ben er samen met Vladimir er even ingedoken. 

We ontdekten dat het breakpoint van 1024px nog niet was uitgewerkt in onze responsive logica. Hierdoor schakelt de layout nu al vanaf 1279px of kleiner over naar de Small Container variant. Dat betekent dat een groot deel van desktop gebruikers de navigatieonderdelen niet direct in beeld krijgt, wat uiteraard niet wenselijk is.

We stellen daarom voor om de overgang naar de Small Container pas te laten plaatsvinden onder de 1024px, zodat we optimaal gebruikmaken van de beschikbare schermruimte op reguliere desktopformaten.

De aangepaste uitwerking vind je terug in het [Schetsboek](https://www.figma.com/design/OnbEtMvfeOF74vXn9ayZnA/Schetsboek---Gemeente-Nijmegen?node-id=403-3208&t=fbeGjDx6vt23empm-11).

**Comment by Rozerinay** [comment]
Het volgende valt me op: wanneer je bijvoorbeeld op Onderwerpen klikt, wordt de tekst bold. Hierdoor verandert de breedte en schuiven de andere menu-items op. Dit wil ik liever voorkomen.

**Comment by vladimirdacosta** [comment]
@mwbankers87 wellicht heb je het al gezien, maar je hebt een paar variaties van de header over het hoofd gezien. Ik mis specifiek de nijmegen-header--mijn-omgeving en de nijmegen-header--funnel.

De mobiele versie van de nijmegen-header--mijn-omgeving gedraagt zich hetzelfde als de standaard mobile header, maar activeert dan de Side Navigation (deze component moet nog ontwikkeld worden).

We kunnen hier maandag even over sparren!

**Comment by mwbankers87** [comment]
@vladimirdacosta @MariannevanLierop De header varianten zijn toegvoegd aan staan op staging.

**Comment by mwbankers87** [comment]
@vladimirdacosta Hover is inmiddels ook opgelost, en staat op staging.

**Comment by vladimirdacosta** [comment]
@Michel-Verhoeven de header is klaar en mag naar release!

**Comment by MariannevanLierop** [comment]
Alleen zoekfunctie werkt nog niet

**Comment by vladimirdacosta** [comment]
@mwbankers87 @Michel-Verhoeven  het is mij opgevallen dat de icoontjes niet goed zijn uitgelijnd. Zouden jullie hier nog even naar willen kijken?

**Comment by mwbankers87** [comment]
@Michel-Verhoeven Dit lijkt ergens niet goed zijn gegaan met de versie op staging. Lokaal zit er een align-items in die op staging niet staat.

**Comment by mwbankers87** [comment]
@vladimirdacosta Bleek een klein linter issue te zijn, en is inmiddels opgelost. Staat op [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/story/html-header--default)

@Michel-Verhoeven Is gepushed naar feat/header-new

cc @MariannevanLierop

**Comment by vladimirdacosta** [comment]
@mwbankers87 het ziet nu beter uit! Top!

**Comment by MariannevanLierop** [comment]
Zodra alle componeten voor search bij elkaar werken (is dat op pro of op staging) zal ik Els vragen het geheel te testen

**Comment by mwbankers87** [comment]
@MariannevanLierop @vladimirdacosta Ik wilde beginnen aan de searchfucntie in de header, maar ik zie geen ontwerp hiervan in Figma. Of staat deze wellicht ergens anders?

**Comment by vladimirdacosta** [comment]
@mwbankers87 Daar gebruiken we het mega-menu voor. We hadden hiervoor alvast een slot (placeholder) ingericht, die later kan worden vervangen door de search.  Zie [link](https://www.figma.com/design/g3AieWAFD3NtkbDDTTFC30/Novio---Nijmegen-Design-System?node-id=6893-191427)

**Comment by mwbankers87** [comment]
@vladimirdacosta Die zocht ik, dank u.

**Comment by mwbankers87** [comment]
@vladimirdacosta Ik heb de search in de [header](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-header--default) gezet.

In Figma was het me niet helemaal duidelijk hoe de uitlijning moest worden. Ik heb hem dus ontwikkeld zoals ik dacht dat het zou moeten worden, zodat je het even kan bekijken. Mocht het niet kloppen, dan kan ik het aanpassen.

Zoeken in het mobile menu komt er aan.

**Comment by vladimirdacosta** [comment]
@mwbankers87 Je hebt helemaal gelijk, ik had de layout nog niet uitgewerkt. De opzet die je nu hebt is niet slecht, hij moet alleen full-width zijn. Ik heb hiervoor een [voorbeeld](
https://www.figma.com/design/LX8k0UfFaMEoD7CJrqtfQm/Templates---Gemeente-Nijmegen?node-id=654-2790) uitgewerkt.

**Comment by mwbankers87** [comment]
@MariannevanLierop @vladimirdacosta Ik heb de search ingebouw conform Figma ontwerp, ook de mobile versie zit er in. Het vergde wel een ombouw van het search component, omdat daarin een maximale breedte was gedefinieerd.

Het webcomponent op de [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-header--default) heeft een werkend voorbeeld.

**Comment by MariannevanLierop** [comment]
Hoi @ElsGroffen, zou jij search ook in de header willen testen op [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-header--default)?

**Comment by vladimirdacosta** [comment]
@mwbankers87 De autocomplete listbox overlapt momenteel de onderlijn van de container waarin het zoekveld staat. Hierdoor valt de listbox visueel over de containergrens heen. De listbox moet binnen de container blijven en de onderlijn van de container moet mee naar beneden schalen wanneer de suggesties zichtbaar zijn, zodat het één samenhangend component vormt.

**Comment by mwbankers87** [comment]
@vladimirdacosta Dat kan gemaakt worden. Maar dan wil ik even informeren naar de [hero](https://wowebnl.github.io/design-system-nijmegen/?path=/story/html-hero--autocomplete). Is het de bedoeling dat dit bij de hero ook gebeurd? Want dan wordt de hero groter naar mate er meer zoekresultaten zijn.

Als de hero wel klopt, betekend dit dat de zoekresultaten zich verschillend gedragen afhankelijk van de parent, soms als 'absolute overlay' en soms als 'static' die ruimte inneemt in de container.

**Comment by vladimirdacosta** [comment]
@mwbankers87 Goede en terechte vraag. Bij de hero is het niet de bedoeling dat de container meegroeit; daar mogen de zoekresultaten als overlay over de hero heen vallen. Excuses dat dit niet eerder duidelijk is gemaakt.

Het klopt dus dat het gedrag afhankelijk is van de parent:
- binnen reguliere containers blijft de listbox binnen de container en schaalt de onderlijn mee
- binnen de hero mag de autocomplete als overlay getoond worden

**Comment by ElsGroffen** [comment]
@mwbankers87 zie eerdere opmerkingen bij #178 en #365

**Comment by vladimirdacosta** [comment]
@mwbankers87 ik mis nog de uitwerking van de taal-switch. Hiervoor wordt de listbox gebruikt. Het ontwerp vind je [hier](https://www.figma.com/design/g3AieWAFD3NtkbDDTTFC30/Novio---Nijmegen-Design-System?node-id=7078-22325).

**Comment by mwbankers87** [comment]
@vladimirdacosta De overlappende listbox in de header search is inmiddels aangepast, zie [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-header--default)

De taal switch volgt volgende week

**Comment by mwbankers87** [comment]
@MariannevanLierop @vladimirdacosta De taal-switch zit nu ook in de header op [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-header--default) Hiermee zou de header compleet moeten zijn.

@Michel-Verhoeven Ik heb voor de header een nieuwe branch aangemaakt, omdat er te veel conflicten zijn met main. De nieuwe branch heet feat/header-new. Hierin zitten dan ook alle sub-modules zoals de search.

**Comment by MariannevanLierop** [comment]
@ElsGroffen, zou jij de header inclusief zoekbalk en taalswitch willen testen op [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-header--default)?

**Comment by vladimirdacosta** [comment]
@mwbankers87 Het ziet er al goed uit. Met een paar kleine aanpassingen kunnen we het helemaal conform het ontwerp maken. Ik heb de [language navigation ](https://www.figma.com/design/OnbEtMvfeOF74vXn9ayZnA/Nijmegen---Schetsboek?node-id=779-106)uitgewerkt om te laten zien hoe deze eruit moet zien en zich moet gedragen.

**Comment by ElsGroffen** [comment]
@mwbankers87 wat mij opvalt is dat de focusrand van de menu-elementen anders is ingericht. Veel minder prominent dan bij de eerdere elementen. Kan dezelfde focusrand gebruikt worden?

Als ik in het zoekveld een stukje tekst invul en dit wil wissen, kan ik het kruisje niet bereiken met alleen het toetsenbord. Het kruisje is aanklikbaar en werkt, maar dus alleen met de muis.

**Comment by mwbankers87** [comment]
@ElsGroffen Klopt, de styling van de focus rand lijkt inderdaad overschreven te worden, ik ga er even naar kijken. Ook of ik het kruisje met tab kan highlighten.

@vladimirdacosta Ik heb even naar het ontwerp gekeken. Op welke punten doel je precies? Ik zie dat hij afwijk qua breedte, maar die minimale breedte komt uit de tokens (--nijmegen-listbox-max-block-size & --nijmegen-listbox-min-inline-size)

Ik kan deze uiteraard overschrijven voor de header, maar dan weet je waar het vandaan komt.

**Comment by vladimirdacosta** [comment]
@mwbankers87Het gaat inderdaad vooral om de positie en de breedte. Kan je  de tokens voor de header overschrijven? Kijk je ook nog even naar de icons? Die lijnen nog steeds niet zo goed!

**Comment by mwbankers87** [comment]
@ElsGroffen De focusrand was inderdaad niet geplaatst, dat is nu gecorrigeerd.
Ook het "wissen" knopje in de search is aangepast, en werkt nu op tab. Zie [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/story/web-component-header--default)

De punten van Vladimir volgen nog.

**Comment by ElsGroffen** [comment]
@mwbankers87 ik zie nu zowel het dunne focusrandje als de gewenste dikke focusrand. Ergens gaat er dus nog iets niet goed.
Ik kan de zoekopdracht nu netjes wissen met het toetsenbord, dat is fijn.

Als ik met enter de zoekbalk activeer, staat de focus niet meteen in het zoekveld. Ik moet hiervoor een extra keer op TAB klikken. Ik zie dat dit op de website van Den Haag ook zo is, dus het zal allemaal wel volgens het design zo kloppen, maar eigenlijk is het niet zo fijn. @vladimirdacosta  wat vind jij hier van?

**Comment by vladimirdacosta** [comment]
@ElsGroffen Klopt. Dit patroon zien we terug op grote platforms zoals GOV.UK en Helsinki. Hier wordt bewust gekozen voor consistentie en voorspelbaarheid boven snelheid. Daarom verplaatsen we de focus niet automatisch: een focus-jump kan voor screenreadergebruikers onverwacht zijn. Voor ervaren keyboardgebruikers voelt dit soms stroever, maar voor andere doelgroepen juist rustiger en duidelijker.

**Comment by ElsGroffen** [comment]
@vladimirdacosta -> dank voor deze info!

**Comment by mwbankers87** [comment]
@ElsGroffen Focusrand is nu zoals hij moet zijn.

@vladimirdacosta Ik heb de pijltjes voor de menu items nu goed gezet (het feit dat dit zolang duurde komt door een Linter issue, die de alignment van het icoon telkens overschreef, dit is nu definitief opgelost)

Ook heb ik de listbox smaller gemaakt voor de header. Echter zie ik in je ontwerp geen pijltjes ervoor staan, deze heeft de listbox standaard wel. Ik weet niet of dit bewust is in Figma. Mochten deze er ook uit moeten, kan ik deze ook overschrijven in de header.

**Comment by vladimirdacosta** [comment]
@mwbankers87 ik moet even mijn excuses aanbieden. Ik zie nu dat we voor de language navigation een andere component hebben gebruikt, namelijk de [dropdown menu](https://www.figma.com/design/g3AieWAFD3NtkbDDTTFC30/Novio---Nijmegen-Design-System?node-id=10979-5141). Daardoor kwam de styling niet overeen. Excuses!

Laten we het voor nu even zo laten en dit oppakken wanneer je daar meer ruimte voor hebt.

**Comment by mwbankers87** [comment]
@vladimirdacosta Dit component toont nog niet goed in de templates, omdat het nog niet is goedgekeurd.
Wil je eerst het dropdown in orde hebben voor deze gereleased kan worden?

**Comment by vladimirdacosta** [comment]
@mwbankers87 dat zou fijn zijn. Scheelt straks weer werk!

**Comment by mwbankers87** [comment]
@MariannevanLierop Zoals besproken heb ik alles zover samengevoegd voor de header. Er staat een PR uit voor Michel, als die is goedgekeurd is het te zien op de template pagina's.
